### PR TITLE
Delete corrupted git repos

### PR DESCRIPTION
**Pull request by kornelski**

### CODE REVIEW

File: crev-lib/src/local.rs [hunk 1]
```diff
@@ -1386,3 +1396,17 @@ fn local_is_send_sync() {
     fn is<T: Send + Sync>() {}
     is::<Local>();
 }
+
+
+fn rm_directory(path_to_delete: &Path) {
```
Comment (dpc) [inline crev-lib/src/local.rs:1401]: I'm always a bit scared seeing such functions. Get we make add some check here, that the directory is a subdirectory (has a prefix) of the dirs we expect to delete stuff in?

**Review by dpc — DISMISSED** [review]

**Comment by dpc** [comment]
What's with you changes never passing `rustfmt` check? :D

**Comment by kornelski** [comment]
I think rustfmt is a deeply flawed tool that often makes code *less* readable, and therefore is unfit to be applied unconditionally to everything.

**Comment by dpc** [comment]
Oooooh. :D

**Comment by thomasjfox** [comment]
> I think rustfmt is a deeply flawed tool that often makes code _less_ readable, and therefore is unfit to be applied unconditionally to everything.

what I love about rustfmt is no more arguing about code style: A project sets it once and can forget about it. Or every organization wide code looks the same, no matter which project. Best invention since sliced bread, IMHO ^^ Also I can write code in the IDE without having to spend brain cycles on indentation / formatting at all and concentrate on the logic. rustfmt runs automatically on save. For me it's a huge productivity win in the end, I can concentrate on the important stuff during code review.

Anyway, on a more serious side:
* when is the git cleanup supposed to run? When I run "crev repo fetch all", I see a lot of repos with github auth failures. I was under the impression the new code should fix this?
* May be it would be good to have a CHANGELOG.md entry for this new feature?

Or is the code just supposed to clean completely corrupted (=borked) git repos?

Current output:
```
$ cargo crev repo fetch all 2>&1 |grep ERROR
ERROR: Error: Failed to fetch https://github.com/AlekSi/crev-proofs: remote authentication required but no callback set; class=Http (34); code=Auth (-16)
ERROR: Error: Failed to fetch https://github.com/vctibor/crev-proofs: remote authentication required but no callback set; class=Http (34); code=Auth (-16)
ERROR: Error: Failed to fetch https://github.com/senden9/crev-proofs: remote authentication required but no callback set; class=Http (34); code=Auth (-16)
ERROR: Error: Failed to fetch https://gitlab.com/SostheneGuedon/crev-proofs: remote authentication required but no callback set; class=Http (34); code=Auth (-16)
ERROR: Error: Failed to fetch https://github.com/victorkoenders/crev-proofs: remote authentication required but no callback set; class=Http (34); code=Auth (-16)
ERROR: Error: Failed to fetch https://github.com/toasteater/crev-proofs: remote authentication required but no callback set; class=Http (34); code=Auth (-16)

**Comment by dpc** [comment]
> `ERROR: Error: Failed to fetch https://github.com/AlekSi/crev-proofs: remote authentication required but no callback set; class=Http (34); code=Auth (-16)`

This error is about not being able to fetch given repository, not that it is corrupted locally.

I'm not sure what to do about it. Probably hide behind some toggle. Seems like it bother people, but on the other hand it is a useful information in case someone was wondering why they can't get data from a certain id/url.

**Comment by kornelski** [comment]
For these I think some kind of exponential backoff would be appropriate. Remember how many fetches failed, and then don't try to fetch them for a while.

**Pull request merged**